File: backend/src/main/java/com/cacodev/shalom/features/event/domain/Event.java
package com.cacodev.shalom.features.event.domain;

import com.cacodev.shalom.common.base.BaseEntity;
import com.cacodev.shalom.features.member.domain.Member;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "EVENTS")
@Entity
public class Event extends BaseEntity {

    private String title;

    @ManyToOne
    private Member organizer;

    private String description;

    private LocalDateTime dateTime;

    private String location;

    @Enumerated(EnumType.STRING)
    private EventStatus status;

    @ManyToMany(fetch = FetchType.EAGER)
    private Set<Member> participants = new HashSet<>();

    private String comments;
}

File: cacodev-ui/src/app/models/Event.ts
import { EventStatus } from "../enums/eventStatus";
import { Participant } from "./Participant";

export interface Event {
    id: string;
    title: string;
    organizerId: string;
    memberId: string;
    description: string;
    dateTime: string;
    location: string;
    status: EventStatus;
    participants: Participant[];
    comments: string;
}